lib/lz4.c

```diff
@@ -515,7 +515,10 @@ static unsigned LZ4_NbCommonBytes (reg_t val)
     assert(val != 0);
     if (LZ4_isLittleEndian()) {
         if (sizeof(val) == 8) {
-#       if defined(_MSC_VER) && (_MSC_VER >= 1800) && defined(_M_AMD64) && !defined(LZ4_FORCE_SW_BITCOUNT)
+#       if defined(_MSC_VER) && (_MSC_VER >= 1800) && defined(_M_AMD64) && defined(__clang__) && !defined(LZ4_FORCE_SW_BITCOUNT)
```

Comment:
minor : I would rather have a dedicated sub-branch `#define __clang__` inside the big compiler test.
It avoids repetition of the big test, and it also more visibly shows that it's a sub-case of the Visual Studio scenario.